File: sdk/node/src/types/applications.ts
export interface Application {
  id: string;
  organization_id?: string;
  name: string;
  slug?: string;
  description?: string;
  app_type?: string;
  client_id: string;
  callback_urls?: string[];
  logout_urls?: string[];
  web_origins?: string[];
  grant_types?: string[];
  allowed_scopes?: string[];
  is_enabled?: boolean;
  is_first_party?: boolean;
  access_token_lifetime_seconds?: number;
  refresh_token_lifetime_seconds?: number;
  created_at?: string;
  updated_at?: string;
}

export interface ApplicationWithSecret extends Application {
  client_secret_plain: string;
}

export interface CreateOAuthAppRequest {
  name: string;
  slug: string;
  app_type: string;
  callback_urls: string[];
  description?: string;
  logo_url?: string;
  logout_urls?: string[];
  web_origins?: string[];
  access_token_lifetime_seconds?: number;
  refresh_token_lifetime_seconds?: number;
  grant_types?: string[];
  allowed_scopes?: string[];
}

export interface UpdateApplicationRequest {
  name?: string;
  description?: string;
  callback_urls?: string[];
  logout_urls?: string[];
  is_enabled?: boolean;
  [key: string]: any;
}

export interface EmailTemplate {
  template_type: string;
  subject?: string;
  html_body?: string;
  text_body?: string;
  variables?: Record<string, any>;
  is_custom?: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface AuthenticateAppRequest {
  client_id: string;
  client_secret: string;
}

File: sdk/python/coreauth/types/applications.py
from __future__ import annotations

from typing import Any, Optional

from pydantic import BaseModel


class CreateApplicationRequest(BaseModel):
    tenant_id: str
    name: str
    description: Optional[str] = None
    application_type: Optional[str] = None
    redirect_uris: list[str] = []
    allowed_scopes: list[str] = []
    metadata: Optional[dict[str, Any]] = None


class Application(BaseModel):
    id: Optional[str] = None
    tenant_id: Optional[str] = None
    organization_id: Optional[str] = None
    name: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    logo_url: Optional[str] = None
    app_type: Optional[str] = None
    client_id: Optional[str] = None
    callback_urls: Optional[list[str]] = None
    logout_urls: Optional[list[str]] = None
    web_origins: Optional[list[str]] = None
    grant_types: Optional[list[str]] = None
    response_types: Optional[list[str]] = None
    allowed_scopes: Optional[list[str]] = None
    is_enabled: Optional[bool] = None
    is_first_party: Optional[bool] = None
    access_token_lifetime_seconds: Optional[int] = None
    refresh_token_lifetime_seconds: Optional[int] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class ApplicationWithSecret(BaseModel):
    id: Optional[str] = None
    tenant_id: Optional[str] = None
    organization_id: Optional[str] = None
    name: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    logo_url: Optional[str] = None
    app_type: Optional[str] = None
    client_id: Optional[str] = None
    callback_urls: Optional[list[str]] = None
    logout_urls: Optional[list[str]] = None
    web_origins: Optional[list[str]] = None
    grant_types: Optional[list[str]] = None
    response_types: Optional[list[str]] = None
    allowed_scopes: Optional[list[str]] = None
    is_enabled: Optional[bool] = None
    is_first_party: Optional[bool] = None
    access_token_lifetime_seconds: Optional[int] = None
    refresh_token_lifetime_seconds: Optional[int] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    client_secret_plain: Optional[str] = None


class CreateOAuthAppRequest(BaseModel):
    name: str
    slug: Optional[str] = None
    description: Optional[str] = None
    logo_url: Optional[str] = None
    app_type: Optional[str] = None
    callback_urls: list[str] = []
    logout_urls: Optional[list[str]] = None
    web_origins: Optional[list[str]] = None
    access_token_lifetime_seconds: Optional[int] = None
    refresh_token_lifetime_seconds: Optional[int] = None
    grant_types: Optional[list[str]] = None
    allowed_scopes: Optional[list[str]] = None
    organization_id: Optional[str] = None


class UpdateApplicationRequest(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    redirect_uris: Optional[list[str]] = None
    allowed_scopes: Optional[list[str]] = None
    is_active: Optional[bool] = None
    metadata: Optional[dict[str, Any]] = None


class EmailTemplate(BaseModel):
    template_type: Optional[str] = None
    subject: Optional[str] = None
    html_body: Optional[str] = None
    text_body: Optional[str] = None
    variables: Optional[list[str]] = None
    is_custom: Optional[bool] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class UpdateEmailTemplateRequest(BaseModel):
    subject: Optional[str] = None
    html_body: Optional[str] = None
    text_body: Optional[str] = None


class AuthenticateAppRequest(BaseModel):
    client_id: str
    client_secret: str
